src/store.ts

```diff
@@ -123,7 +123,7 @@ export function useStore(selector, options: SelectorOptions = {}): any {
       shouldRender = fn() || shouldRender;
     }
     if (shouldRender) {
-      return component.render();
+      return component.render(true);
```

Comment:
As per discussed, what I would like in this case is to force the render of the current component when it's observed state changes (the current diff should be good for that), but at the same time I don't want children component to be force-rendered too. I want the children to be rendered only if that makes sense according to their own `shouldUpdate()` value, or if their own observed state changes.

Reply:
Btw just to be clear, I'm not saying children should never render in this context, just that their `shouldUpdate()` should be respected: no render if it returns `false`, but since the default value is `true`, they will still be rendered in normal cases.
This gives the responsibility to whoever is overriding that method to ensure it returns the proper value according to what needs to be done.